src/core/agent/services/ipexCommunicationService.ts

```diff
@@ -320,31 +320,33 @@ class IpexCommunicationService extends AgentService {
     return op;
   }
 
-  async markAcdcComplete(credentialId: string) {
+  async markAcdc(
+    credentialId: string,
+    status: CredentialStatus.CONFIRMED | CredentialStatus.REVOKED
+  ) {
     const metadata = await this.credentialStorage.getCredentialMetadata(
       `metadata:${credentialId}`
     );
-    if (!metadata) {
-      throw new Error(
-        IpexCommunicationService.CREDENTIAL_MISSING_METADATA_ERROR_MSG
+    if (metadata) {
```

Comment:
Not sure why we removed the `CREDENTIAL_MISSING_METADATA_ERROR_MSG` error here?

Reply:
This `markAcdc` is being called in a `while (true)` loop; if we throw it, the loop might crash. So I think this will be better. What do you think?

Reply:
No that's OK - that shouldn't change the behaviour. It's on my list of items to make those `while (true)` loops recover in case of errors. For now this should be reverted